code/lightna/magento-os-frontend/js/common/AddToCart.js

```diff
@@ -1,35 +1,38 @@
-import { Cookie } from '../lib/Cookie';
 import { UserInput } from '../lib/UserInput';
 import request from '../lib/HttpClient';
-import { $, $$ } from "../lib/utils/dom";
+import { $ } from "../lib/utils/dom";
 
 export class AddToCart {
-    components = [];
     addToCartUrl = '/checkout/cart/add';
     classes = {
         loading: 'loading',
         disabled: 'btn-disabled',
     };
 
     constructor() {
-        this.components = $$('.cjs-add-to-cart');
+        this.component = '.cjs-add-to-cart';
+        this.trigger = '[data-action="add-to-cart"]';
         this.bindEvents();
     }
 
     bindEvents() {
-        this.components.forEach((component) => {
-            const actionTrigger = $('[data-action="add-to-cart"]', component);
-            actionTrigger.addEventListener('click', () => {
-                this.addProduct(component, actionTrigger);
-                this.toggleAnimation(actionTrigger, true);
-            });
+        $('body').addEventListener('click', (event) => {
```

Comment:
I remember we discussed it, could you remind what it simplifies?

Reply:
we do not create instances for each component; once new content lazy loaded there is no need to init add-to-cart for it/add event listener

Reply:
that's good, but I'm thinking if it's really scalable...

Reply:
take your time